evm-ds/src/precompiles/scilla_call.rs

```diff
@@ -0,0 +1,132 @@
+use evm::backend::Backend;
+use evm::executor::stack::{PrecompileFailure, PrecompileOutput, PrecompileOutputType};
+use evm::{Context, ExitError};
+use std::borrow::Cow;
+
+use crate::precompiles::scilla_common::get_contract_addr_and_name;
+use crate::precompiles::scilla_common::substitute_scilla_type_with_sol;
+use ethabi::decode;
+use ethabi::param_type::ParamType;
+use ethabi::token::Token;
+use hex::ToHex;
+use serde_json::{json, Value};
+
+const BASE_COST: u64 = 15;
+const PER_BYTE_COST: u64 = 3;
+
+pub(crate) fn scilla_call(
+    input: &[u8],
+    gas_limit: Option<u64>,
+    _contex: &Context,
+    backend: &dyn Backend,
+    _is_static: bool,
+) -> Result<(PrecompileOutput, u64), PrecompileFailure> {
+    let gas_needed = match required_gas(input) {
+        Ok(i) => i,
+        Err(err) => return Err(PrecompileFailure::Error { exit_status: err }),
+    };
+
+    if let Some(gas_limit) = gas_limit {
+        if gas_limit < gas_needed {
+            return Err(PrecompileFailure::Error {
+                exit_status: ExitError::OutOfGas,
+            });
+        }
+    }
+
+    let (code_address, passed_transition_name) = get_contract_addr_and_name(input)?;
+
+    let code = backend.code_as_json(code_address);
+    if code.is_empty() {
+        return Err(PrecompileFailure::Error {
```

Comment:
Why is this an error? (OK, it's an error if this isn't a contract, perhaps) - a call to an empty contract should presumably behave just as though you were calling an ordinary contract, but trying to invoke a transition which doesn't exist - which, it seems, just returns empty?